lib/OpenQA/Task/Job/ArchiveResults.pm

```diff
@@ -11,6 +11,7 @@ sub register ($self, $app, @args) {
 }
 
 sub _archive_results ($minion_job, @args) {
+    my $ensure_task_retry_on_termination_signal_guard = OpenQA::Task::SignalGuard->new($minion_job);
```

Comment:
That's quite the name. 🥇

Reply:
The name was suggested by @okurz in a previous PR review. I'm just using it consistently here as well.